### PR TITLE
Update ghworkflow actions to use pre-commit

### PR DESCRIPTION
**Pull request by JGSweets**
Adds pre-commit to gh workflow.

### CODE REVIEW

**Comment by JGSweets** [comment]
@taylorfturner looks like when all is ready, this will work.

**Comment by taylorfturner** [comment]
@JGSweets looks like the pre-commit failed re-formatted and fixed some things ...

File: .github/workflows/test-python-package.yml [hunk 1]
```diff
@@ -24,17 +24,15 @@ jobs:
...
       run: |
         python -m pip install --upgrade pip
-        python -m pip install flake8 pytest
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
         if [ -f requirements-ml.txt ]; then pip install -r requirements-ml.txt; fi
+        if [ -f requirements-dev.txt ]; then pip install -r requirements-dev.txt; fi
         if [ -f requirements-test.txt ]; then pip install -r requirements-test.txt; fi
         if [ -f requirements-reports.txt ]; then pip install -r requirements-reports.txt; fi
-    - name: Lint with flake8
+    - name: run pre-commit
```
Comment (JGSweets) [inline .github/workflows/test-python-package.yml:32]: now runs full pre-commit

File: dataprofiler/data_readers/data.py [hunk 2]
```diff
@@ -26,7 +27,13 @@ class Data(object):
         dict(data_class=TextData, kwargs=dict()),
     ]
 
-    def __new__(cls, input_file_path: Optional[str]=None, data: Optional[Any]=None, data_type: Optional[str]=None, options: Optional[Dict]=None):
```
Comment (JGSweets) [inline dataprofiler/data_readers/data.py:29]: formatting

File: dataprofiler/labelers/base_model.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 import copy
 import inspect
 import warnings
-from typing import Any, Dict, Iterator, List, Optional, Tuple, Type, Union, cast
```
Comment (JGSweets) [inline dataprofiler/labelers/base_model.py:8]: unused import

File: dataprofiler/profilers/data_labeler_column_profile.py [hunk 4]
```diff
@@ -287,7 +287,9 @@ def label_representation(self) -> Optional[Dict[str, float]]:
         if not self.sample_size:
             return None
 
-        label_representation: Dict[str, float] = {key: 0 for key in self.possible_data_labels}
```
Comment (JGSweets) [inline dataprofiler/profilers/data_labeler_column_profile.py:290]: format

File: dataprofiler/reports/graphs.py [hunk 5]
```diff
@@ -22,14 +22,18 @@
     # installed
     pass
 
-from dataprofiler.profilers.profile_builder import StructuredColProfiler, StructuredProfiler
+from dataprofiler.profilers.profile_builder import (
```
Comment (JGSweets) [inline dataprofiler/reports/graphs.py:25]: format

File: dataprofiler/reports/graphs.py [hunk 6]
```diff
@@ -182,7 +186,9 @@ def plot_col_histogram(
 
 @utils.require_module(["matplotlib", "seaborn"])
 def plot_missing_values_matrix(
-    profiler: StructuredProfiler, ax: Optional[matplotlib.axes.Axes] = None, title: Optional[str] = None
```
Comment (JGSweets) [inline dataprofiler/reports/graphs.py:185]: format

File: dataprofiler/reports/graphs.py [hunk 7]
```diff
@@ -204,7 +210,9 @@ def plot_missing_values_matrix(
 
 @utils.require_module(["matplotlib", "seaborn"])
 def plot_col_missing_values(
-    col_profiler_list: List[StructuredColProfiler], ax: Optional[matplotlib.axes.Axes] = None, title: Optional[str] = None
```
Comment (JGSweets) [inline dataprofiler/reports/graphs.py:207]: format

**Review by taylorfturner — APPROVED** [review]

**Review by tyfarnan — APPROVED** [review]

**Pull request merged**